cmd/istioctl/mixer.go

```diff
@@ -205,28 +240,81 @@ func mixerRuleGet(scope, subject string) (string, error) {
 	return string(data), nil
 }
 
-func mixerRuleDelete(scope, subject string) error {
+func mixerRequest(method, path string, reqBody []byte) error {
+	status, respBody, err := mixerRESTRequester.Request(method, path, reqBody)
 
-	path := mixerRulePath(scope, subject)
-	status, body, err := mixerRESTRequester.Request(http.MethodDelete, path, nil)
-	if body != nil {
+	// If we got output, let's look at it, even if we got an error.  The output might include the reason for the error.
+	if respBody != nil {
 		var response mixerAPIResponse
 		message := "unknown"
-		if errJSON := json.Unmarshal(body, &response); errJSON == nil {
+		if errJSON := json.Unmarshal(respBody, &response); errJSON == nil {
 			message = response.Status.Message
 		}
 
 		if status != http.StatusOK {
-			return fmt.Errorf("failed rule deletion with status %v: %s", status, message)
+			return fmt.Errorf("failed rule creation with status %v: %s", status, message)
```

Comment:
`mixerRequest` is also called by `mixerRuleDelete` in which case this error message is incorrect. Maybe just generalize this with the following?

```golang
return fmt.Errorf("request failed with status: %v: %v", status, message)
```